File: backend/app/schemas/player.py
from pydantic import BaseModel, Field


class CurrentTrackResponse(BaseModel):
    spotify_id: str
    name: str
    artist_names: list[str] = Field(
        default_factory=list,
    )
    album_name: str
    album_image_url: str | None = None
    duration_ms: int
    spotify_url: str | None = None


class CurrentPlaybackResponse(BaseModel):
    is_active: bool
    is_playing: bool
    progress_ms: int | None = None
    track: CurrentTrackResponse | None = None

File: frontend/src/api/player.ts
import { apiRequest } from './client'

export interface CurrentTrack {
  spotify_id: string
  name: string
  artist_names: string[]
  album_name: string
  album_image_url: string | null
  duration_ms: number
  spotify_url: string | null
}

export interface CurrentPlaybackResponse {
  is_active: boolean
  is_playing: boolean
  progress_ms: number | null
  track: CurrentTrack | null
}

export function getCurrentPlayback(
  signal?: AbortSignal,
): Promise<CurrentPlaybackResponse> {
  return apiRequest<CurrentPlaybackResponse>(
    '/api/player/current',
    { signal },
  )
}